v2/bolt_store.go

```diff
@@ -51,6 +53,12 @@ type Options struct {
 	// write to the log. This is unsafe, so it should be used
 	// with caution.
 	NoSync bool
+
+	// MsgpackUseNewTimeFormat is used to force the underlying msgpack codec to
+	// use the newer format of time.Time when encoding, used in versions <=0.5.5
+	// by default. Decoding is not affected, as all decoders know how to decode
+	// both formats.
+	MsgpackUseNewTimeFormat bool
```

Comment:
If this is the format from 0.5.5 and before, shouldn't this be MsgpackUse **Old** TimeFormat? I swear I'm not trying to be pedantic and am genuinely confused. :sweat_smile:

Reply:
`MsgpackUseNewTimeFormat` being `true` will cause `TimeNotBuiltin` to be set to `false` when configuring the msgpack Codec. 

`TimeNotBuiltin` forces the msgpack library to not treat time as a builtin but instead rely on the binary marshaller (or text marshaller interfaces according to the docs).

I think the behavior is correct and also that having the zero value here cause disabling of the not-backwards-compatible time encoding is best so that we don't break Consul or Nomad.

Reply:
@mkeeler is correct. I got the comment slightly wrong, and will correct it.

It's confusing due to the double-negative and trying to work with the default-`false` behavior of Go.